README.md

```diff
@@ -10,26 +10,222 @@ Distributed Consensus Algorithms often take the form of a replicated state machi
+	
+	* `initial_state` is called when Raft is initialized. This interface will return a `RaftState` which contains `HardState` and `ConfState`;
+	
+	    * `HardState` contains the last meta information including commit index, the vote leader, and the vote term;
+	
+	    * `ConfState` records the current node IDs like `[1, 2, 3]` in the cluster. Every Raft node must have a unique ID in the cluster; 
+	
+	* `entries` returns the Log entries in an interval `[low, high)`;
+	
+	* `term` returns the term of the entry at Log index;
+	
+	* `first_index` and `last_index` return the first and last index of the Log;
+	
+	Pay attention to what is returned when there is no Log but it needs to get the `term` at index `first_index() - 1`. To solve this, you can use a dummy Log entry to keep the last truncated Log entry. See [`entries: vec![Entry::new()]`](src/storage.rs#L85) as a reference.
+	
+	* The last interface is `snapshot`, which returns a Snapshot of the current state machine. This Snapshot data will be sent to another node.
+
+2. Configure the Raft node
+
+	After creating the Raft Storage, use `RawNode::new()` to create the Raft node. The `new` function needs a Raft Storage and a configuration. The following fields are important for this configuration:
+	
+	* `id`: the unique ID of the node in the cluster, which must be unique; 
+	
+	* `election_tick`: how many ticks the follower re-campaigns if it doesn’t receive any message from the leader;
+	
+	* `heartbeat_tick`: how many ticks the leader sends the heartbeat to the followers to keep alive;
+	
+	* `applied`: the last applied index for the state machine. Raft will resume applying Log entries to the state machine from this index; 
+	
+	* `max_size_per_msg`: Raft can send many Log entries at the same time, so we need to limit the maximum size of the sending message. This is an optimization for Transport in batch;
+	
+	* `max_inflight_msgs`: how many messages the leader can send to the followers without acknowledgement. This is an optimization for the Transport in pipeline.
+	
+	* `election_tick` must be larger than `heartbeat_tick`. If our tick interval is 100 ms, we can use 10 for `election_tick` and 3 for `heartbeat_tick`, which means the leader will send heartbeat to the followers every 300 ms and the follower will re-campaign without receiving any messages after 1 second. 
+	
+	* The `read_only_option` enables you to choose the linearizability mode or the lease mode to read data. If you don’t care about the read consistency and want a higher read performance, you can use the lease mode.
+	
+	Other important fields like `check_quorum` and `pre_vote` are used to avoid the disturbance and make the cluster more stable. 
+	
+### Step 2: Drive and Run the Raft node
+
+1. Use a timer to run the Raft node regularly. See the following example for using Rust channel `recv_timeout`:
+
+    ```rust
+    let mut t = Instant::now();
```

Comment:
It would be good to show how `r` comes into existence here.

Also, please avoid using single character variable names in examples. `t -> time`, `r -> raft`, `d -> duration`.